### PR TITLE
Support Consistent Address Deployment Across Networks

### LINKED ISSUE #131: Smart Contracts Not Actually Deployed using Create3 with Foundry to Network
## Description and context

When attempting to verify smart contracts that are deployed using the Create3 library with Foundry, the contracts cannot be verified on testnets. This appears to be due to Foundry not sending actual transactions when deploying contracts with the Create3 library.

## Steps to reproduce
1. Deploy a smart contract using Main.s.sol with Foundry on a testnet.
2. Attempt to verify the contract on the testnet.
3. Notice the failure in verification due to contract not deployed actaully.

## Experienced behavior
Currently, contracts deployed using the Create3 library via Foundry can only be verified in local test environments. Verification fails on real testnets because actual deployment transactions are not sent.

## Expected behavior
Smart contracts deployed using the Create3 should be deployed on both local environments and real testnets.

## Solution recommendation
To address this issue, the `Create3Deployer` should be deployed as a standalone contract. This would ensure that the deployment involves an actual transaction, enabling the verification of the contract on testnets.

### PR DESCRIPTION
**Pull request by kingster-will**
### Overview
This PR introduces an update to the smart contract deployment process using Foundry and the `CREATE3`. By deploying a standalone `Create3Deployer` contract on networks (e.g., Sepolia), we enable Foundry to execute actual transactions through `Create3Deployer`, and ensuring that contracts are deployed at the same deterministic addresses.

### Key Features
- **Consistent Address Across Chains:** Supports deploying contracts at the same address across all chains by using the same `salt` value derived from a provided `seed`. This ensures uniformity and predictability of contract addresses on various networks.
- **Enhanced Deployment Script:** The deployment script now accepts a `seed` argument. This `seed` is used to generate the `salt` for the Create3 deployment, ensuring that using the same `seed` on different chains will deploy contracts to the same address.

### Motivation
Previously, deploying smart contracts using Foundry with the Create3 library faced verification issues on testnets due to the absence of actual deployment transactions. This PR resolves this issue by ensuring that each deployment involves a real transaction on the blockchain.

### Additional Benefits
This update not only fixes the verification issue but also enhances the flexibility and control we have over the deployment process. It allows for seamless address changes without sacrificing consistency across multiple chains.

### Usage
Deploy contracts by running the deployment script with a specific `seed`:
```bash
forge script script/foundry/deployment/Main.s.sol:Main <seed value e.g. 8> --sig "run(uint256)" --fork-url <rpc url> -vvvv --broadcast  --verify
```

This command will ensure that the same contract addresses are used across different networks, provided the same `seed` is used.

## Test Plan 
- Deploy on Sepolia testnet and verified all contracts.
- Repeat deployment using the same `seed` on different networks and verify the addresses match.

## Related Issue
Closes #131 

## Note
**This PR also include the change in another PR #115  of licensing hook, will rebase after the PR#115 merged.**

### CODE REVIEW

**Review by LeoHChen — APPROVED** [review]
Please rebase after previous PR merged. Changes in this PR lgtm.

File: script/foundry/utils/DeployHelper.sol [hunk 1]
```diff
@@ -546,12 +662,12 @@ contract DeployHelper is Script, BroadcastManager, JsonDeploymentHandler, Storag
 
     /// @dev get the salt for the contract deployment with CREATE3
     function _getSalt(string memory name) private view returns (bytes32 salt) {
-        salt = keccak256(abi.encode(name, block.number));
+        salt = keccak256(abi.encode(name, create3SaltSeed));
```
Comment (Spablob) [inline script/foundry/utils/DeployHelper.sol:665]: Informational only - with this change we will need a way to keep track of the previous salt seeds used before running the script

**Review by Spablob — APPROVED** [review]

**Comment by jdubpark** [comment]
Appreciate the thorough PR readme!

File: script/foundry/utils/DeployHelper.sol (hunk 1 above)
Comment (kingster-will) [reply script/foundry/utils/DeployHelper.sol:665]: Agreed. The deployment would fail if we deploy contracts again with the same seed on the same chain.

**Pull request merged**